### ISSUE TITLE
cannot range over 3 (untyped int constant)compilerInvalidRangeExpr

### ISSUE THREAD

**Issue by kkchu791** [mmcgrana/gobyexample#544]
In the For section, there's this code
```
    for i := range 3 {
        fmt.Println("range", i)
    }
```

It seems you can't iterate over an integer, you can only use range for `array, slice, string, map, or channel.`

**Comment by eliben** [comment]
https://github.com/mmcgrana/gobyexample?tab=readme-ov-file#what-version-of-go-is-required-to-run-these-examples

**Comment by kkchu791** [comment]
Would it be best to just remove the examples with `range <int>`?

**Comment by eliben** [comment]
> Would it be best to just remove the examples with `range <int>`?

This feature was released in Go 1.22, about 6 months ago. We expect gobyexample to be mostly used for folks who are learning the language, and these people don't have a reason to use old versions. Moreover, people turn to gobyexample specifically to find out how new features work in concrete small code examples. Therefore, we try to keep examples up to date with the newest language features.
